### ISSUE TITLE
Render inconsistencies with three backticks (code block) (compared to GitHub)

### ISSUE THREAD

**Issue by isoos** [dart-lang/markdown#259]
Original report on pub site: https://github.com/dart-lang/pub-dev/issues/2697

Root cause:
```
print(markdownToHtml(' * ``` a ``` b'''));
// prints: 
// <ul><li>
// <pre><code class="language-a"></code></pre></li></ul>
//
// note: the "b" disappearing, and "a" is converted to language annotation
//
// expected:
// <ul><li><code>a</code> b</li></ul>
```

GitHub renders it as the user expected:
https://github.com/DokkarRachidReda/Fancy-Flutter-Alert-Dialog#use-the-package

**Comment by jonasfj** [comment]
I would think expected output is:
```
// <ul><li><pre><code>a</code></pre> b</li></ul>
```
But I could be wrong, we probably need to refer to a spec for markdown, if one exists :)

**Comment by jonasfj** [comment]
See: https://spec.commonmark.org/0.29/#code-spans

> A code span begins with a backtick string and ends with a backtick string of equal length.

It seems `package:markdown` has issues with 3 or more backticks.

**Comment by srawlins** [comment]
Fix pushed in markdown 2.1.0.
